src/jit/codegenxarch.cpp

```diff
@@ -4989,6 +4989,20 @@ void CodeGen::genCallInstruction(GenTreePtr node)
 
 #endif // defined(_TARGET_X86_)
 
+#ifdef FEATURE_AVX_SUPPORT
+    // When it's a PInvoke call and the call type is USER function, we issue VZEROUPPER here
+	// if the function contains 256bit AVX instructions, this is to avoid AVX-256 to Legacy SSE
+	// transition penalty, assume the user function may contains legacy SSE instruction
+    if (call->IsPInvoke() && call->gtCallType == CT_USER_FUNC && compiler->getFloatingPointInstructionSet() == InstructionSet_AVX)
+    {
+        if (getEmitter()->Contains256bitAVX())
+        {
```

Comment:
Note that Lower register specification detecting the usage of AVX 128/256-bit instructions also solves the problem that Carol pointed out above, because Lower Regiter specification runs before Codegen phase.

The method that can set a flag to indicate whether AVX 128/256-bit instructions will be used is lowerxarch.cpp!Lowering:TreeNodeInfoInitSIMD()